src/test/java/com/google/api/codegen/testdata/go_main_library.baseline

```diff
@@ -790,14 +790,86 @@ func (it *StringIterator) takeBuf() interface{} {
 }
 
 
+// BookOperation manages a long-running operation yielding librarypb.Book.
 type BookOperation struct {
-    *longrunning.Operation
+    lro *longrunning.Operation
 }
 
+// BookOperation returns a new BookOperation from a given name.
+// The name must refer to a long-running operation created by the same service.
+func (c *Client) BookOperation(name string) *BookOperation {
+    return &BookOperation{
+        lro: longrunning.InternalNewOperation(c.Connection(), &longrunningpb.Operation{Name: name}),
+    }
+}
+
+// Wait blocks until the long-running operation is completed, returning the response and any errors encountered.
+//
+// See documentation of Poll for error-handling information.
 func (op *BookOperation) Wait(ctx context.Context) (*librarypb.Book, error) {
     var resp librarypb.Book
-    if err := op.Operation.Wait(ctx, &resp); err != nil {
+    if err := op.lro.Wait(ctx, &resp); err != nil {
         return nil, err
     }
     return &resp, nil
 }
+
+// Poll fetches the latest state of the long-running operation.
+//
+// If Poll fails, the error is returned and op is unmodified. If Poll succeeds and
+// the operation has completed with failure, the error is returned and op.Done will return true.
+// If Poll succeeds and the operation has completed successfully,
+// op.Done will return true, and the response of the operation is returned.
+// If Poll succeeds and the operation has not completed, the returned response and error are both nil.
+func (op *BookOperation) Poll(ctx context.Context) (*librarypb.Book, error) {
+    var resp librarypb.Book
+    if err := op.lro.Poll(ctx, &resp); err != nil {
+        return nil, err
+    }
+    if !op.Done() {
+        return nil, nil
+    }
+    return &resp, nil
+}
+
+// Metadata returns metadata associated with the long-running operation.
+// Latest metadata can be fetched by Poll.
```

Comment:
It doesn't. I was trying to document something like
> Metadata itself doesn't contact the server, but Poll does.
> So, if you need the latest metadata, call Poll first to fetch it into the object,
> then call Metadata to get it out.

I've moved this section of the doc into Poll. Please let me know if it reads better to you.